Dockerfile

```diff
@@ -1,20 +1,24 @@
-FROM alpine:3.8 AS build
+FROM ubuntu:18.04 AS build
 WORKDIR /go/src/github.com/prebid/prebid-server/
-RUN apk add -U --no-cache go git dep musl-dev
+RUN \
+     sed -i 's/# deb/deb/g' /etc/apt/sources.list && \
```

Comment:
**The above line will presumably remove uncomment all lines with "# deb"**
Yes, from the original `/etc/apt/sources.list` file that comes with the image. Not all of the deb lines are originally not commented out. If we uncomment them all we have access to more repositories.

Reply:
**It occurs to me that it may be better to make sure the pattern is anchored at the beginning of the line (i.e., "^# deb") to avoid uncommenting any real comments that may trail actual code on a line**
It's not necessary, the file is pretty small and the regular expression "# deb" only matches repository URLs. File /etc/apt/sources.list AFTER sed edition:
```
root@2794e056a135:/# cat /etc/apt/sources.list
# See http://help.ubuntu.com/community/UpgradeNotes for how to upgrade to
# newer versions of the distribution.
deb http://archive.ubuntu.com/ubuntu/ bionic main restricted
deb-src http://archive.ubuntu.com/ubuntu/ bionic main restricted

## Major bug fix updates produced after the final release of the
## distribution.
deb http://archive.ubuntu.com/ubuntu/ bionic-updates main restricted
deb-src http://archive.ubuntu.com/ubuntu/ bionic-updates main restricted

## N.B. software from this repository is ENTIRELY UNSUPPORTED by the Ubuntu
## team. Also, please note that software in universe WILL NOT receive any
## review or updates from the Ubuntu security team.
deb http://archive.ubuntu.com/ubuntu/ bionic universe
deb-src http://archive.ubuntu.com/ubuntu/ bionic universe
deb http://archive.ubuntu.com/ubuntu/ bionic-updates universe
deb-src http://archive.ubuntu.com/ubuntu/ bionic-updates universe

## N.B. software from this repository is ENTIRELY UNSUPPORTED by the Ubuntu
## team, and may not be under a free licence. Please satisfy yourself as to
## your rights to use the software. Also, please note that software in
## multiverse WILL NOT receive any review or updates from the Ubuntu
## security team.
deb http://archive.ubuntu.com/ubuntu/ bionic multiverse
deb-src http://archive.ubuntu.com/ubuntu/ bionic multiverse
deb http://archive.ubuntu.com/ubuntu/ bionic-updates multiverse
deb-src http://archive.ubuntu.com/ubuntu/ bionic-updates multiverse

## N.B. software from this repository may not have been tested as
## extensively as that contained in the main release, although it includes
## newer versions of some applications which may provide useful features.
## Also, please note that software in backports WILL NOT receive any review
## or updates from the Ubuntu security team.
deb http://archive.ubuntu.com/ubuntu/ bionic-backports main restricted universe multiverse
deb-src http://archive.ubuntu.com/ubuntu/ bionic-backports main restricted universe multiverse

## Uncomment the following two lines to add software from Canonical's
## 'partner' repository.
## This software is not part of Ubuntu, but is offered by Canonical and the
## respective vendors as a service to Ubuntu users.
deb http://archive.canonical.com/ubuntu bionic partner
deb-src http://archive.canonical.com/ubuntu bionic partner

deb http://security.ubuntu.com/ubuntu/ bionic-security main restricted
deb-src http://security.ubuntu.com/ubuntu/ bionic-security main restricted
deb http://security.ubuntu.com/ubuntu/ bionic-security universe
deb-src http://security.ubuntu.com/ubuntu/ bionic-security universe
deb http://security.ubuntu.com/ubuntu/ bionic-security multiverse
deb-src http://security.ubuntu.com/ubuntu/ bionic-security multiverse
```

Reply:
**I'm guessing these commands are all run every time the file is executed.**
Yes
**Is it safe and necessary to always execute all these commands? In particular, I'm wondering about the commands on line 9.**
It's not uncommon for a Dockerfile to have these commands. If any of these fail, the entire build fails, and a proper error code is returned. I'd say it is as safe as the previous version

Reply:
Line 9 cleans the cache and temp folders that got populated from the package update and installation instructions above

Reply:
Do we know why the sources come commented out by default and need to be uncommented?

Reply:
The `/etc/apt/sources.list` file itself discusses it bruefly. This is the original version of the file, before the `sed` command on our Dockerfile kicks in
```
root@c23c1615c75b:/# cat /etc/apt/sources.list
# See http://help.ubuntu.com/community/UpgradeNotes for how to upgrade to
# newer versions of the distribution.
deb http://archive.ubuntu.com/ubuntu/ bionic main restricted
# deb-src http://archive.ubuntu.com/ubuntu/ bionic main restricted

## Major bug fix updates produced after the final release of the
## distribution.
deb http://archive.ubuntu.com/ubuntu/ bionic-updates main restricted
# deb-src http://archive.ubuntu.com/ubuntu/ bionic-updates main restricted

## N.B. software from this repository is ENTIRELY UNSUPPORTED by the Ubuntu
## team. Also, please note that software in universe WILL NOT receive any
## review or updates from the Ubuntu security team.
deb http://archive.ubuntu.com/ubuntu/ bionic universe
# deb-src http://archive.ubuntu.com/ubuntu/ bionic universe
deb http://archive.ubuntu.com/ubuntu/ bionic-updates universe
# deb-src http://archive.ubuntu.com/ubuntu/ bionic-updates universe

## N.B. software from this repository is ENTIRELY UNSUPPORTED by the Ubuntu
## team, and may not be under a free licence. Please satisfy yourself as to
## your rights to use the software. Also, please note that software in
## multiverse WILL NOT receive any review or updates from the Ubuntu
## security team.
deb http://archive.ubuntu.com/ubuntu/ bionic multiverse
# deb-src http://archive.ubuntu.com/ubuntu/ bionic multiverse
deb http://archive.ubuntu.com/ubuntu/ bionic-updates multiverse
# deb-src http://archive.ubuntu.com/ubuntu/ bionic-updates multiverse

## N.B. software from this repository may not have been tested as
## extensively as that contained in the main release, although it includes
## newer versions of some applications which may provide useful features.
## Also, please note that software in backports WILL NOT receive any review
## or updates from the Ubuntu security team.
deb http://archive.ubuntu.com/ubuntu/ bionic-backports main restricted universe multiverse
# deb-src http://archive.ubuntu.com/ubuntu/ bionic-backports main restricted universe multiverse

## Uncomment the following two lines to add software from Canonical's
## 'partner' repository.
## This software is not part of Ubuntu, but is offered by Canonical and the
## respective vendors as a service to Ubuntu users.
# deb http://archive.canonical.com/ubuntu bionic partner
# deb-src http://archive.canonical.com/ubuntu bionic partner

deb http://security.ubuntu.com/ubuntu/ bionic-security main restricted
# deb-src http://security.ubuntu.com/ubuntu/ bionic-security main restricted
deb http://security.ubuntu.com/ubuntu/ bionic-security universe
# deb-src http://security.ubuntu.com/ubuntu/ bionic-security universe
deb http://security.ubuntu.com/ubuntu/ bionic-security multiverse
# deb-src http://security.ubuntu.com/ubuntu/ bionic-security multiverse
```

Reply:
Let me double check again whether or not what we are installing is available in the default repository URL's (it wasn't according to many tests I performed, but let me look further into it because maybe we don't need to uncomment them all)

Reply:
PR updated

Reply:
Going back to my comment about line 9:  my concern was that we're nuking the contents of /tmp and /var/tmp which are public directories that could also contain data from other programs, users, etc.  How can we be sure that randomly nuking those directories is safe?

Reply:
**/var/lib/apt/lists**
According to [https://unix.stackexchange.com/questions/217369/clear-apt-get-list](https://unix.stackexchange.com/questions/217369/clear-apt-get-list)
*rm /var/lib/apt/lists/\* will remove the package lists. No repositories will be deleted, all that can happen is that tools like apt-cache cannot get package information and apt-get install will fail unless you run apt-get update again*

**/var/tmp and /tmp**
Both /var/tmp and /tmp are short term and long term temporary files folders. In our context, the temporary files stored there so far are those of the package installation process and won't be relevant again because installation won't be run again until a new container is created from scratch. Up to line 6, *prebid-server* has not even run for the first time so no relevant files for its operation will be there.